crates/driver/src/domain/mempools.rs

```diff
@@ -34,6 +35,37 @@ impl Mempools {
             .boxed()
         })));
     }
+
+    /// Get gas price that is used for executing the settlement.
+    /// Since there are several mempools with different gas price estimators, we
+    /// will get them all and then pick the highest one.
+    pub async fn gas_price(&self, settlement: &Settlement) -> anyhow::Result<eth::GasPrice> {
+        let mut gas_prices = Vec::new();
+        for mempool in &self.0 {
+            let gas_price_estimator = mempool.gas_price_estimator(settlement);
+            let gas_price = gas_price_estimator.estimate().await?;
+            gas_prices.push(gas_price);
+        }
+
+        gas_prices.sort_unstable_by(|a, b| {
```

Comment:
You could turn this into `max_by()` to avoid any confusion on sorting order (i.e. are small element left or right).